### PR TITLE
New Options

### PR DESCRIPTION
**Pull request by Ferks-FK**
This PR adds some of the options suggested in #299.

## Todo
- [x] Block the registration of new users.
- [x] Block the creation of new servers.
- [ ] Disable Login.

Any idea of another option? :)

### CODE REVIEW

**Comment by sybetjeh** [comment]
with the disable login, also log everyone out if you enable that (otherwie what would be the point?) and maybe add a whitelist

**Comment by Ferks-FK** [comment]
> with the disable login, also log everyone out if you enable that (otherwie what would be the point?) and maybe add a whitelist

I also see no real reason to disable the login.

**Comment by sybetjeh** [comment]
Maintenance? Panel for only yourself? Depends what you're gonna use it for

**Comment by Ferks-FK** [comment]
> 

What kind of maintenance can users not login?
There must also be artisan maintenance.

File: app/Http/Controllers/Api/UserController.php [hunk 1]
```diff
@@ -251,6 +251,13 @@ public function store(Request $request)
             'email' => ['required', 'string', 'email', 'max:64', 'unique:users'],
             'password' => ['required', 'string', 'min:8', 'max:191'],
         ]);
+
```
Comment (Ferks-FK) [inline app/Http/Controllers/Api/UserController.php:254]: Make it impossible to create users via the API if the option is disabled on the front-end.

File: database/seeders/Seeds/SettingsSeeder.php [hunk 2]
```diff
@@ -518,21 +518,21 @@ public function run()
         Settings::firstOrCreate([
             'key'   => 'SETTINGS::SYSTEM:ENABLE_UPGRADE',
         ], [
-            'value' =>"",
-            'type'  => 'string',
+            'value' =>  "false",
```
Comment (Ferks-FK) [inline database/seeders/Seeds/SettingsSeeder.php:521]: Change the type of a key in the db, and default values for the new options.

File: app/Http/Controllers/Api/UserController.php [hunk 3]
```diff
@@ -251,6 +251,13 @@ public function store(Request $request)
             'email' => ['required', 'string', 'email', 'max:64', 'unique:users'],
             'password' => ['required', 'string', 'min:8', 'max:191'],
         ]);
+        
+        // Prevent the creation of new users via API if this is enabled.
+        if (!config('SETTINGS::SYSTEM:CREATION_OF_NEW_USERS', 'true')) {
+            throw ValidationException::withMessages([
+                'error' => "The creation of new users has been blocked by the system administrator."
+            ]);
+        }
```
Comment (Ferks-FK) [inline app/Http/Controllers/Api/UserController.php:260]: Make it impossible to create users via the API if the option is disabled on the front-end.

File: database/seeders/Seeds/SettingsSeeder.php [hunk 4]
```diff
@@ -518,21 +518,21 @@ public function run()
...
         ]);
         Settings::firstOrCreate([
-            'key'   => 'SETTINGS::SYSTEM:DISABLE_NEW_USERS',
+            'key'   => 'SETTINGS::SYSTEM:CREATION_OF_NEW_USERS',
         ], [
-            'value' => "",
+            'value' => "true",
             'type'  => 'boolean',
             'description'  => 'Enable/disable the creation of new users'
         ]);
```
Comment (Ferks-FK) [inline database/seeders/Seeds/SettingsSeeder.php:538]: Change the type of a key in the db, and default values for the new options.

**Pull request merged**